src/lua/log.lua

```diff
@@ -127,13 +133,16 @@ end
 
 -- Default options. The keys are part of
 -- user API , so change with caution.
-local log_cfg = {
+local default_cfg = {
     log             = nil,
     nonblock        = nil,
     level           = S_INFO,
     format          = fmt_num2str[ffi.C.SF_PLAIN],
 }
 
+local log_cfg = table.copy(default_cfg)
+local log
```

Comment:
Nit: Looks like this `log` forward declaration is required solely to use `log.debug` in the functions below. If so, it'd be better to define `log_debug` local function instead.

Reply:
Done.